iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/InjectionObservabilityImpl.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Injection;
+import com.powsybl.iidm.network.extensions.InjectionObservability;
+import org.jgrapht.alg.util.Pair;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class InjectionObservabilityImpl<T extends Injection<T>> extends AbstractExtension<T>
+        implements InjectionObservability<T> {
+
+    private boolean observable;
+
+    private final Pair<Double, Boolean> standardDeviationP;
+
+    private final Pair<Double, Boolean> standardDeviationQ;
+
+    private final Pair<Double, Boolean> standardDeviationV;
+
+    public InjectionObservabilityImpl(T component, boolean observable,
+                                      double standardDeviationP, boolean redundantP,
+                                      double standardDeviationQ, boolean redundantQ,
+                                      double standardDeviationV, boolean redundantV) {
+        super(component);
+        this.observable = observable;
+        this.standardDeviationP = new Pair<>(standardDeviationP, redundantP);
```

Comment:
I guess in the network overall it will not make a HUGE difference, but it would be more memory efficient to keep primitive types instead of the pairs, although they are logically linked.
Keep in mind that we will have it on all elements of the network, though!